model/version_db.go

```diff
@@ -280,8 +280,11 @@ func UpdateVersionProject(versionID string, versionNumber int, pp *ParserProject
 	}
 	return errors.Wrap(VersionUpdateOne(
 		bson.M{
-			VersionIdKey:           versionID,
-			VersionConfigNumberKey: versionNumber,
+			VersionIdKey: versionID,
+			"$or": []bson.M{
+				bson.M{VersionConfigNumberKey: bson.M{"$exists": false}},
+				bson.M{VersionConfigNumberKey: versionNumber},
```

Comment:
This prevents an update in the case where something else has updated the project and updated the versionNumber. If I understand this correctly, should `UpdateVersionProject` only run if the version number is 0 or empty? Do you know if this will happen if the version number is set to something greater than 0? My concern is that it's possible fo two `UpdateVersionProject`s to race if the version number is non-empty, since this doesn't increment the version number.

Reply:
This isn't if the version number is 0, it's if the version number is equal to the version number that our current version has (i.e. if the config hasn't been updated). I think we're safe from a race because:

1) If UpdateVersionProject was to somehow update twice (because the version number is empty or the number is equal), it would set parserProject twice, but using the same config (given that the number hasn't changed) so this wouldn't give bad data.

2) If something else were to actually change the config string/parser project, then the update number would have been incremented, and so we won't match with a version.